### PR TITLE
SONARJAVA-900 RSPEC-2440 Class with only static methods instantiation check

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  @Override
+  public void visitNode(Tree tree) {
+    NewClassTree newClassTree = (NewClassTree) tree;
+    if (hasOnlyStaticMethod((AbstractTypedTree) newClassTree.identifier())) {
+      String className = ((IdentifierTree) newClassTree.identifier()).name();
+      addIssue(tree, "Remove this instantiation of \"" + className + "\".");
+    }
+  }
+
+  private boolean hasOnlyStaticMethod(AbstractTypedTree tree) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java]: I would pass the symbol or the NewClassTree not the AbstractTypedTree...

File: java-checks/src/main/java/org/sonar/java/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  @Override
+  public List<Kind> nodesToVisit() {
+    return ImmutableList.of(Kind.NEW_CLASS);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    NewClassTree newClassTree = (NewClassTree) tree;
+    if (hasOnlyStaticMethod((AbstractTypedTree) newClassTree.identifier())) {
+      String className = ((IdentifierTree) newClassTree.identifier()).name();
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java]: ClassTree identifier is not necessarily an identifier.

File: java-checks/src/main/java/org/sonar/java/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      addIssue(tree, "Remove this instantiation of \"" + className + "\".");
+    }
+  }
+
+  private boolean hasOnlyStaticMethod(AbstractTypedTree tree) {
+    Collection<Symbol> members = tree.getSymbolType().getSymbol().members().scopeSymbols();
+    boolean hasOnlyThisOrSuper = true;
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java]: should be named hasStaticMethod and inversed

File: java-checks/src/main/java/org/sonar/java/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    for (Symbol symbol : members) {
+      if (!isThisOrSuper(symbol)) {
+        if (!isStaticMethod(symbol)) {
```
Comment (benzonico) [inline java-checks/src/main/java/org/sonar/java/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java]: Condition should be inverted

File: java-checks/src/test/files/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
+class A {
```
Comment (benzonico) [inline java-checks/src/test/files/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java]: Maybe adding the case of a static inner class.

Comment (m-g-sonar) [inline java-checks/src/test/files/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java]: Inner classes can not be static. See [jls 8.1.3](http://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.1.3).

**Comment by benzonico** [comment]
LGTM!

**Pull request closed without merging**